docs/source/daml/intro/daml/daml-intro-3/Keys.daml

```diff
@@ -60,30 +62,43 @@ id_ref_test = scenario do
       amount = 100.0
 
   accountCid <- submit accountant do
-     create Account with
+     createCmd Account with
       accountant
       owner
       bank
       number = "ABC123"
 
+  Some account <- queryContractId accountant accountCid
   balanceCid <- submit accountant do
-    account <- fetch accountCid
-    create CashBalance with
+    createCmd CashBalance with
       accountant
       cash
       account = key account
 
   -- Now the accountant updates the telephone number for the bank on the account
+  Some account <- queryContractId accountant accountCid
   new_accountCid <- submit accountant do
-    account <- fetch accountCid
-    archive accountCid
-    create account with
+    archiveCmd accountCid
+    cid <- createCmd account with
       bank = account.bank with
         telephone = "098 7654 321"
+    pure cid
 
   -- Thanks to contract keys, the current account contract is fetched
-  submit accountant do
-    balance <- fetch balanceCid
-    (cid, account) <- fetchByKey @Account balance.account
-    assert (cid == new_accountCid)
+  Some balance <- queryContractId accountant balanceCid
+  (cid, account) <- submit accountant do
+    createAndExerciseCmd (Helper accountant) (FetchAccountByKey balance.account)
+  assert (cid == new_accountCid)
+
+-- Helper template to call `fetchByKey`.
+template Helper
```

Comment:
@bame-da I’ve removed all the other `Helper` templates but this one. We could in principle add a `queryContractKey` function to remove this as well but I’m not sure this is a good idea:

1. For keys, it matters that you fetch within the same transaction since the contents associated with the key can change.
2. We don’t use this anywhere else. I think it’s crucial to actually introduce `fetchByKey` so you have to introduce `fetchByKey` as well as `queryContractId` which just adds more stuff to learn.

Reply:
I'm slightly on the fence, since you'd usually call queryByKey on the JSON API or similar, but let's not hold the PR over this.

Reply:
Opened https://github.com/digital-asset/daml/issues/7340 for this. As mentioned below, I think this is useful but we cannot slot it right in here without other changes since `fetchByKey` should be explained somewhere in the daml intro imho.